File: admin-base-api/Admin.WebApi/Models/DashboardModels.cs
namespace Admin.WebApi.Models
{
    public class DashboardSummaryDto
    {
        public decimal TotalSalesThisMonth { get; set; }
        public decimal TotalSalesAllTime { get; set; }
        public decimal PendingAmount { get; set; }
        public decimal AvailableAmount { get; set; }
        public DateTime? NextAvailableAt { get; set; }
        public decimal TotalWithdrawn { get; set; }
        public int PhotosSoldThisMonth { get; set; }
        public int TotalPhotosSold { get; set; }
        public int ActiveEventsCount { get; set; }
    }

    public class EventSalesDto
    {
        public int EventId { get; set; }
        public string EventName { get; set; } = string.Empty;
        public string ProductType { get; set; } = string.Empty;
        public decimal TotalSales { get; set; }
        public int PhotosSold { get; set; }
        public decimal PendingAmount { get; set; }
        public decimal AvailableAmount { get; set; }
    }

    public class SaleDetailDto
    {
        public int SaleId { get; set; }
        public DateTime SoldAt { get; set; }
        public string EventName { get; set; } = string.Empty;
        public string ProductType { get; set; } = string.Empty;
        public string BuyerName { get; set; } = string.Empty;
        public string? BuyerEmail { get; set; }
        public string PaymentMethod { get; set; } = string.Empty;
        public decimal TotalAmount { get; set; }
        public int Quantity { get; set; }
        public string Status { get; set; } = string.Empty;
        public string? ExternalReference { get; set; }
        public List<SalePurchasedPhotoDto> PurchasedPhotos { get; set; } = new();
    }

    public class SalePurchasedPhotoDto
    {
        public int PhotoId { get; set; }
        public string Label { get; set; } = string.Empty;
    }
}

File: capturar-front/src/app/features/panel/data-access/dashboard.models.ts
export interface DashboardSummaryDto {
  totalSalesThisMonth: number;
  totalSalesAllTime: number;
  pendingAmount: number;
  availableAmount: number;
  nextAvailableAt?: string | null;
  totalWithdrawn: number;
  photosSoldThisMonth: number;
  totalPhotosSold: number;
  activeEventsCount: number;
}

export interface EventSalesDto {
  eventId: number;
  eventName: string;
  productType: string;
  totalSales: number;
  photosSold: number;
  pendingAmount: number;
  availableAmount: number;
}

export interface SaleDetailDto {
  saleId: number;
  soldAt: string;
  eventName: string;
  productType: string;
  buyerName: string;
  buyerEmail?: string | null;
  paymentMethod: string;
  totalAmount: number;
  quantity: number;
  status: string;
  externalReference?: string | null;
  purchasedPhotos: SalePurchasedPhotoDto[];
}

export interface SalePurchasedPhotoDto {
  photoId: number;
  label: string;
}

export interface ApproveTransferResultDto {
  success: boolean;
  message: string;
  externalReference?: string | null;
  status?: string;
}

export interface LiquidationDto {
  liquidationId: number;
  liquidationDate: string;
  amount: number;
  fromDate: string;
  toDate: string;
  ordersCount: number;
}

export interface WithdrawalResultDto {
  success: boolean;
  message: string;
  amount: number;
  processedAt: string;
}
